internal/services/qumulo/qumulo_file_system_resource.go

```diff
@@ -0,0 +1,274 @@
+		},
+
+		"availability_zone": commonschema.ZoneSingleOptionalForceNew(),
+
+		"location": commonschema.Location(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r FileSystemResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r FileSystemResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 180 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Qumulo.FileSystemsClient
+
+			var config FileSystemResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			id := filesystems.NewFileSystemID(subscriptionId, config.ResourceGroupName, config.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			payload := filesystems.FileSystemResource{
+				Location: location.Normalize(config.Location),
+				Tags:     tags.Expand(config.Tags),
+				Properties: filesystems.FileSystemResourceProperties{
+					AdminPassword:     config.AdminPassword,
+					AvailabilityZone:  pointer.To(config.AvailabilityZone),
+					DelegatedSubnetId: config.DelegatedSubnetId,
+					InitialCapacity:   config.InitialCapacity,
+					StorageSku:        filesystems.StorageSku(config.StorageSku),
+					UserDetails: filesystems.UserDetails{
+						Email: config.UserEmailAddress,
+					},
+					MarketplaceDetails: filesystems.MarketplaceDetails{
+						OfferId:     offerId,
+						PlanId:      config.MarketplacePlanId,
+						PublisherId: publisherId,
+					},
+				},
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, payload); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r FileSystemResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Qumulo.FileSystemsClient
+			schema := FileSystemResourceSchema{}
+
+			id, err := filesystems.ParseFileSystemID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var config FileSystemResourceSchema
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+			schema.AdminPassword = config.AdminPassword
+			schema.UserEmailAddress = config.UserEmailAddress
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
```

Comment:
if it's gone, we can exit this function early without pulling the config from the state etc, can we move this up?